dataland-frontend/tests/e2e/specs/high/Prepopulation.ts

```diff
@@ -12,56 +19,91 @@ describe('Population Test', () => {
 
     });
 
+    async function uploadData(dataArray:Array<object>, endpoint:string){
+        const start = Date.now()
+        const chunkSize = 80;
+        for (let i = 0; i < dataArray.length; i += chunkSize) {
+            const chunk = dataArray.slice(i, i + chunkSize);
+            await Promise.all(chunk.map(async (element:object) => {
+                    await fetch(`${Cypress.env("API")}/${endpoint}`, {
+                        method: 'POST',
+                        headers: {
+                            'Content-Type': 'application/json'
+                        },
+                        body: JSON.stringify(element)
+                    }).then(response => {
+                        assert(response.status.toString() === "200" )
+                    })
+                })
+            )
+        }
+        const millis = Date.now() - start
+        console.log(`seconds elapsed = ${Math.floor(millis / 1000)}`)
+    }
 
-    it('Populate Companies', function (){
-        for (const index in companiesData) {
-            cy.request('POST', `${Cypress.env("API")}/companies`, companiesData[index]).its('status').should("equal", 200)
 
-        }
-        console.log(companiesData)
+    it('Populate Companies',  async() => {
+        await uploadData(companiesData, "companies")
```

Comment:
Eigentlich braucht man hier kein await mehr, oder?

Reply:
du hast recht. In einer normalen Code bräuchte man nicht. aber cypress macht es bisschen anders. Es gibt manchmal Probleme wenn man innerhalb cypress awaits benutzt wiel in cypress "Commands are enqueued and run asynchronously":
https://stackoverflow.com/questions/57228703/how-to-fix-warning-cypress-detected-that-you-returned-a-promise-in-a-test
https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Return-Values

aber was wir festgestellt haben ist so: wenn man in einer von implemetierten Stellen kein awiat macht, wartet cypress nicht auf response und lässt die später in einer Queue laufen. 
Falls die Erklärung nicht deutlich war, können wir telefonieren

Reply:
passt so.